modules/openapi-generator/src/main/java/org/openapitools/codegen/utils/ModelUtils.java

```diff
@@ -1213,18 +1248,63 @@ public static Schema getSchemaFromResponse(OpenAPI openAPI, ApiResponse response
      * @param content a 'content' section in the OAS specification.
      * @return the Schema.
      */
-    private static Schema getSchemaFromContent(Content content) {
+    private static Schema getFirstSchemaFromContent(Content content) {
+        MediaType mediaType = getFirstMediaTypeFromContent(content);
+
+        if (mediaType == null) {
+            return null;
+        } else {
+            return mediaType.getSchema();
+        }
+    }
+
+    /**
+     * Return the first MediaType from a specified OAS 'content' section.
+     * <p>
+     * For example, given the following OAS, this method returns the 'application/json'
+     * media type because it is listed first in the OAS.
+     * <p>
+     * responses:
+     *   '200':
+     *     content:
+     *       application/json:
+     *         schema:
+     *           $ref: '#/components/schemas/XYZ'
+     *       application/xml:
+     *          ...
+     *
+     * @param content a 'content' section in the OAS specification.
+     * @return the Schema.
+     */
+    private static MediaType getFirstMediaTypeFromContent(Content content) {
         if (content == null || content.isEmpty()) {
             return null;
         }
+
         Map.Entry<String, MediaType> entry = content.entrySet().iterator().next();
+
         if (content.size() > 1) {
             // Other content types are currently ignored by codegen. If you see this warning,
             // reorder the OAS spec to put the desired content type first.
             once(LOGGER).debug("Multiple schemas found in the OAS 'content' section, returning only the first one ({})",
                     entry.getKey());
         }
-        return entry.getValue().getSchema();
+
+        return entry.getValue();
+    }
+
+    private static Map<String, Schema> getSchemasFromContent(Content content) {
```

Comment:
minor suggestion: add a doc string explaining what this function does (even it's pretty self-explanatory)